snippets.py

```diff
@@ -274,7 +276,9 @@ def get(self):
             category = email_to_category.get(
                 snippet.email, models.NULL_CATEGORY
             )
-            snippets_by_category.setdefault(category, []).append(snippet)
+            snippets_by_category.setdefault(category, []).append(
```

Comment:
You're right, if a user changes their display-name all their previous snippets should also change, so we don't want to *just* denormalize onto the snippet.

But I don't want the display-name to go away if the user is deleted, I don't think.  So maybe we should do what you suggest, but backwards:
1. If the snippet's user's display-name is non-empty, show user's display-name,
2. Else, if the snippet's display-name is non-empty, show snippet's display-name,
3. Else, just show the email.

> splitting snippets_and_users_by_category into two variables will make code a bit harder to write

Whichever way you think ends up being cleaner code is good by me.

> Also if you want to change category's behavior as well.

Yeah, I can see an argument for that.  It would also make things more consistent.  But probably a separate PR (maybe after this one is done? so you can just follow the pattern you used here).